common.scad

```diff
@@ -289,13 +289,28 @@ function get_anchor(anchor,center,uncentered=BOT,dflt=CENTER) =
 //   d = Most general diameter.
 //   dflt = Value to return if all other values given are `undef`.
 function get_radius(r1=undef, r2=undef, r=undef, d1=undef, d2=undef, d=undef, dflt=undef) = (
-    !is_undef(r1)? assert(is_undef(r2)&&is_undef(d1)&&is_undef(d2), "Conflicting or redundant radius/diameter arguments given.") r1 :
-    !is_undef(r2)? assert(is_undef(d1)&&is_undef(d2), "Conflicting or redundant radius/diameter arguments given.") r2 :
-    !is_undef(d1)? d1/2 :
-    !is_undef(d2)? d2/2 :
-    !is_undef(r)? assert(is_undef(d), "Conflicting or redundant radius/diameter arguments given.") r :
-    !is_undef(d)? d/2 :
-    dflt
+    !is_undef(r1)
```

Comment:
Good idea. Yet, `num_defined()` doesn't call `sum()` but it has 2 main issues:
1. it would break if parameter `v` isn't a list;
2. it would break if an user call it with more than 1 parameter (this is common to many other recursive functions).

The former is solved by an `assert`.  To solve the later we have 3 alternatives:
1. call an internal recursive function to do the job;
2. change it to a C-like for avoiding recursion;
3. call `sum()` which is fast and non-recursive for vectors.

I prefer the later.

Reply:
Good idea.